internal/types/container/keepalived_container_collector_host.go

```diff
@@ -135,12 +140,41 @@ func (k *KeepalivedContainerCollectorHost) sigNum(sigString string) syscall.Sign
 
 // Signal sends signal to Keepalived process.
 func (k *KeepalivedContainerCollectorHost) signal(signal syscall.Signal) error {
-	err := k.dockerCli.ContainerKill(context.Background(), k.containerName, strconv.Itoa(int(signal)))
+	data, err := os.ReadFile(k.pidPath)
 	if err != nil {
-		logrus.WithError(err).WithField("signal", int(signal)).Error("Failed to send signal")
+		logrus.WithField("path", k.pidPath).WithError(err).Error("Can't find keepalived pid. Use default kill command")
```

Comment:
I guess this fallback would be annoying to be logged here and somehow under control - maybe better to have it as an option and also not falling back on every error?

Reply:
This modification requires users to add a new mount at "/var/run/keepalived.pid". 
To ensure compatibility with previous versions, I’m keeping a fallback. 
If someone always uses the 'latest' tag for the Docker image and doesn’t notice this change, the new version will still work with their existing keepalived container.